### DISCUSSION TITLE
Idea for improved workflow for CTS and hold cell congestion

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3795] [General]
What if CTS as a default made a moderate effort to repair hold violations with an option to try harder and then continued on to detailed routing?

This would would give the user a result to examine that has hold violations. The user can debug the problem looking at setup and hold timing in the GUI and via OpenSTA commands.

My experience is that if hold violations aren't easily fixed(small), then there is a fundamental underlying problem that has to be fixed.

Debugging global route problems is much tougher and more time consuming than looking up some hold violations in the GUI.

Thoughts?

**Comment by maliberty** [comment]
You can always stop the flow after cts if you want.  I'm not sure what is proposed here.  You can't go from cts to detailed routing without going through global routing.

**Reply by oharboe** [reply]
If CTS or global routing fails(or worse runs "forever") due to hold cells, how do you debug it?

Not easy.... requires a strong mental model of what is going on and tools mastery.

if you have hold or setup failures: easy to debug from the gui.
